### PR TITLE
CI: add benchmark workflow

### PR DESCRIPTION
**Pull request by trym-b**
Best reviewed commit by commit

### CODE REVIEW

**Comment by trym-b** [comment]
This is how the benchmark will look like: https://trym-b.github.io/warchaeology/benchmarks/ubuntu-22.04/

**Comment by trym-b** [comment]
This is how a benchmark failure will look like: https://github.com/trym-b/warchaeology/actions/runs/8201485917

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -35,6 +35,40 @@ jobs:
...
+      matrix:
+        os: [ubuntu-22.04]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+          go-version: "1.21"
+      - name: Run benchmark
+        run: go test ./... -bench="Bench" -count 10 | tee benchmark-result.txt
```
Comment (maeb) [inline .github/workflows/main.yml:50]: Maybe use default number of iterations

File: .github/workflows/publish_benchmarks.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Code Checkout
+        uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: "1.21"
+      - name: Run benchmarks
+        run: go test ./... -bench="Bench" -count 10 | tee benchmark-result.txt
```
Comment (maeb) [inline .github/workflows/publish_benchmarks.yml:30]: Maybe use default number of iterations

File: .github/workflows/main.yml (hunk 1 above)
Comment (trym-b) [reply .github/workflows/main.yml]: Fixed in fixup commit

File: cmd/cat/cat_test.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import (
+	"testing"
+	"time"
+)
+
+func BenchmarkDummy(b *testing.B) {
+	// This is a dummy test, it should be replaced with something more
+	// meaningful in a later commit
+	for i := 0; i < b.N; i++ {
+		time.Sleep(1 * time.Second)
```
[the remark this answers is not in the document]
Comment (trym-b) [reply cmd/cat/cat_test.go]: Fixed in fixup commit

File: .github/workflows/publish_benchmarks.yml (hunk 2 above)
Comment (trym-b) [reply .github/workflows/publish_benchmarks.yml]: Fixed in fixup commit

**Comment by trym-b** [comment]
Ready for next round @maeb

**Review by maeb — APPROVED** [review]

**Pull request merged**